himbaechel/uarch/gowin/gowin.cc

```diff
@@ -7,230 +7,234 @@
 
 NEXTPNR_NAMESPACE_BEGIN
 
-void GowinImpl::init(Context *ctx) {
-	h.init(ctx);
-	HimbaechelAPI::init(ctx);
-	// These fields go in the header of the output JSON file and can help
-	// gowin_pack support different architectures
-	ctx->settings[ctx->id("packer.arch")] = std::string("himbaechel/gowin");
-	// XXX it would be nice to write chip/base name in the header as well,
-	// but maybe that will come up when there is clarity with
-	// Arch::archArgsToId
+void GowinImpl::init(Context *ctx)
+{
+    h.init(ctx);
+    HimbaechelAPI::init(ctx);
+    // These fields go in the header of the output JSON file and can help
+    // gowin_pack support different architectures
+    ctx->settings[ctx->id("packer.arch")] = std::string("himbaechel/gowin");
+    // XXX it would be nice to write chip/base name in the header as well,
+    // but maybe that will come up when there is clarity with
+    // Arch::archArgsToId
 }
 
-void GowinImpl::prePlace() {
-	ctx->cells.at(ctx->id("leds_OBUF_O"))->setAttr(ctx->id("BEL"), std::string("X0Y14/IOBA"));
-	ctx->cells.at(ctx->id("leds_OBUF_O_1"))->setAttr(ctx->id("BEL"), std::string("X0Y15/IOBB"));
-	ctx->cells.at(ctx->id("leds_OBUF_O_2"))->setAttr(ctx->id("BEL"), std::string("X0Y20/IOBB"));
-	ctx->cells.at(ctx->id("leds_OBUF_O_3"))->setAttr(ctx->id("BEL"), std::string("X0Y21/IOBB"));
-	ctx->cells.at(ctx->id("leds_OBUF_O_4"))->setAttr(ctx->id("BEL"), std::string("X0Y24/IOBB"));
-	ctx->cells.at(ctx->id("leds_OBUF_O_5"))->setAttr(ctx->id("BEL"), std::string("X0Y25/IOBB"));
-	ctx->cells.at(ctx->id("rst_IBUF_I"))->setAttr(ctx->id("BEL"), std::string("X0Y4/IOBA"));
-	assign_cell_info();
+void GowinImpl::prePlace()
+{
+    ctx->cells.at(ctx->id("leds_OBUF_O"))->setAttr(ctx->id("BEL"), std::string("X0Y14/IOBA"));
```

Comment:
I know this is test code, but testing with multiple designs would probably be easier manually instantiating the IO buffers and then setting the bel attribute in the Verilog

Reply:
I know :) will be fixed